io.openems.edge.evcs.ocpp.server/src/io/openems/edge/evcs/ocpp/server/MyJsonServer.java

```diff
@@ -122,10 +139,16 @@ public void lostSession(UUID sessionIndex) {
 				for (Entry<String, UUID> session : MyJsonServer.this.parent.ocppSessions.entrySet()) {
 					if (session.getValue().equals(sessionIndex)) {
 						ocppId = session.getKey();
+						break;
 					}
 				}
 
-				MyJsonServer.this.parent.ocppSessions.remove(ocppId);
+				// Before removing, check if the sessionIndex is still the one mapped to ocppId
```

Comment:
This code also deals with the window time between establishing the new session and discarding the old one inside `newSession`, so it might not be necessary after all. During our weekend tests, we couldn’t hit this code path without artificially introducing sleeps, which suggests we may not need it.